commands/compile/compile.go

```diff
@@ -33,26 +34,48 @@ import (
 	"github.com/arduino/arduino-cli/legacy/builder/i18n"
 	"github.com/arduino/arduino-cli/legacy/builder/types"
 	rpc "github.com/arduino/arduino-cli/rpc/commands"
+	"github.com/arduino/arduino-cli/telemetry"
 	paths "github.com/arduino/go-paths-helper"
 	properties "github.com/arduino/go-properties-orderedmap"
+	"github.com/segmentio/stats/v4"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 )
 
 // Compile FIXMEDOC
 func Compile(ctx context.Context, req *rpc.CompileReq, outStream, errStream io.Writer, debug bool) (*rpc.CompileResp, error) {
+
+	tags := map[string]string{
+		"fqbn":            req.Fqbn,
+		"sketchPath":      telemetry.SanitizeSketchPath(req.SketchPath),
+		"showProperties":  strconv.FormatBool(req.ShowProperties),
+		"preprocess":      strconv.FormatBool(req.Preprocess),
+		"buildProperties": strings.Join(req.BuildProperties, ","),
+		"warnings":        req.Warnings,
+		"verbose":         strconv.FormatBool(req.Verbose),
+		"quiet":           strconv.FormatBool(req.Quiet),
+		"vidPid":          req.VidPid,
+		"exportFile":      req.ExportFile,
+		"jobs":            strconv.FormatInt(int64(req.Jobs), 10),
+		"libraries":       strings.Join(req.Libraries, ","),
+		"success":         "false",
+	}
+
 	pm := commands.GetPackageManager(req.GetInstance().GetId())
 	if pm == nil {
+		stats.Incr("compile", stats.M(tags)...)
```

Comment:
Can we `defer` this at the begginig? it's more future-proof IMHO since it's very easy to forget to call `stats.Incr("compile", stats.M(tags)...)` before returning.